### PR TITLE
[docsite] updating templates for docsite site

### PR DESCRIPTION
**Pull request by cesrjimenez**
This PR focuses on updating our templates in response to the recent changes made in `sourcegraph/docsite` in this [PR](https://github.com/sourcegraph/docsite/pull/96). The intention behind these updates is to ensure alignment with the latest changes and design standards provided by the Sourcegraph documentation team.

## Test plan
Tested manually and added tests.

### CODE REVIEW

File: doc/_resources/templates/root.html [hunk 1]
```diff
@@ -9,16 +9,22 @@
 		<head>
 			<title>{{block "title" .}}Home{{end}} - Sourcegraph docs</title>
             <link rel="icon" type="image/png" href="https://about.sourcegraph.com/sourcegraph-mark.png" />
-			<link rel="stylesheet" type="text/css" href="{{asset "docsite.css"}}" />
+            {{if .ContentVersion}}
+                <link rel="stylesheet" type="text/css" href="{{assetsFromVersion .ContentVersion "docsite.css" }}" />
+                <script src="{{assetsFromVersion .ContentVersion "docsite.js" }}"></script>
+                <script src="{{assetsFromVersion .ContentVersion "railroad.js" }}"></script>
```
Comment (mucles) [inline doc/_resources/templates/root.html:15]:
```suggestion
                <script src="{{.assetsFromVersion .ContentVersion "docsite.js" }}"></script>
                <script src="{{.assetsFromVersion .ContentVersion "railroad.js" }}"></script>
```

File: doc/_resources/templates/root.html [hunk 2]
```diff
@@ -9,16 +9,22 @@
 		<head>
 			<title>{{block "title" .}}Home{{end}} - Sourcegraph docs</title>
             <link rel="icon" type="image/png" href="https://about.sourcegraph.com/sourcegraph-mark.png" />
-			<link rel="stylesheet" type="text/css" href="{{asset "docsite.css"}}" />
+            {{if .ContentVersion}}
+                <link rel="stylesheet" type="text/css" href="{{assetsFromVersion .ContentVersion "docsite.css" }}" />
```
Comment (mucles) [inline doc/_resources/templates/root.html:13]:
```suggestion
                <link rel="stylesheet" type="text/css" href="{{assetsFromVersion .ContentVersion "docsite.css" }}" />
```

File: doc/_resources/templates/root.html (hunk 1 above)
Comment (mucles) [inline doc/_resources/templates/root.html:15]:
```suggestion
                <link rel="stylesheet" type="text/css" href="{{.assetsFromVersion .ContentVersion "docsite.css" }}" />
                <script src="{{.assetsFromVersion .ContentVersion "docsite.js" }}"></script>
                <script src="{{.assetsFromVersion .ContentVersion "railroad.js" }}"></script>
```

File: doc/_resources/templates/root.html [hunk 3]
```diff
@@ -177,8 +182,13 @@
 			<aside id="sidebar">
                 <header>
 				    <h1 id="logo"><a href="/">
-                        <img src="{{asset "logo-theme-light.svg"}}" class="theme-light" alt="Sourcegraph docs"/>
-                        <img src="{{asset "logo-theme-dark.svg"}}" class="theme-dark" alt="Sourcegraph docs"/>
+                    {{if .ContentVersion}}
+                        <img src="{{assetsFromVersion .ContentVersion "logo-theme-light.svg" }}" class="theme-light" alt="Sourcegraph docs"/>
+                        <img src="{{assetsFromVersion .ContentVersion "logo-theme-dark.svg" }}" class="theme-dark" alt="Sourcegraph docs"/>
```
Comment (mucles) [inline doc/_resources/templates/root.html:187]:
```suggestion
                        <img src="{{.assetsFromVersion .ContentVersion "logo-theme-light.svg" }}" class="theme-light" alt="Sourcegraph docs"/>
                        <img src="{{.assetsFromVersion .ContentVersion "logo-theme-dark.svg" }}" class="theme-dark" alt="Sourcegraph docs"/>
```

**Review by mucles — CHANGES_REQUESTED** [review]
The following changes should resolve the Bazel error, tested locally. 
image

File: doc/_resources/templates/root.html (hunk 1 above)
Comment (Numbers88s) [reply doc/_resources/templates/root.html:15]: I'm not sure that is the reason. The `assetsFromVersion` is a template function much like this one https://github.com/sourcegraph/sourcegraph/pull/55681/files#diff-ca039c482acee552fabec35e6aefe88310ebb6cd3f9bcbe97643340f37df4c29R236

**Comment by Numbers88s** [comment]
> The following changes should resolve the Bazel error, tested locally. image

Interesting, then how come [this template function](https://github.com/sourcegraph/sourcegraph/pull/55681/files#diff-ca039c482acee552fabec35e6aefe88310ebb6cd3f9bcbe97643340f37df4c29R236) doesn't need the dot?

**Comment by mucles** [comment]
> @mucles Interesting, then how come [this template function](https://github.com/sourcegraph/sourcegraph/pull/55681/files#diff-ca039c482acee552fabec35e6aefe88310ebb6cd3f9bcbe97643340f37df4c29R236) doesn't need the dot?
@Numbers88s Oh, you're right. I thought it was a variable. My mistake; that is old that it would pass.  You have this [PR](https://github.com/sourcegraph/docsite/pull/96) where you defined **assetsFromVersion**. This most likely will need to be merged first before the test will pass.

**Review by efritz — APPROVED** [review]

**Review by mucles — APPROVED** [review]

**Pull request merged**